library/core/src/slice/mod.rs

```diff
@@ -3479,6 +3479,155 @@ impl<T> [T] {
 
         left
     }
+
+    /// Returns multiple references at once, without doing bounds checking.
+    ///
+    /// For a safe alternative see [`get_many`].
+    ///
+    /// # Safety
+    ///
+    /// Calling this method with overlapping or out-of-bounds indices is *[undefined behavior]*
+    /// even if the resulting references are not used.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// #![feature(get_many)]
+    ///
+    /// let x = &[1, 2, 4];
+    ///
+    /// unsafe {
+    ///     assert_eq!(x.get_many_unchecked([0, 2]), [&1, &4]);
+    /// }
+    /// ```
+    ///
+    /// [`get_many`]: slice::get_many
+    /// [undefined behavior]: https://doc.rust-lang.org/reference/behavior-considered-undefined.html
+    #[unstable(feature = "get_many", issue = "none")]
+    #[inline]
+    pub unsafe fn get_many_unchecked<const N: usize>(&self, indices: [usize; N]) -> [&T; N] {
+        // NB: We need to go through a raw pointer to make miri happy
+        let slice: *const [T] = self;
+        let mut arr: mem::MaybeUninit<[&T; N]> = mem::MaybeUninit::uninit();
+        let arr_ptr = arr.as_mut_ptr();
+
+        // SAFETY: We expect `indices` to contain disjunct values that are
+        // in bounds of `self`.
+        unsafe {
+            for i in 0..N {
+                let idx = *indices.get_unchecked(i);
+                *(*arr_ptr).get_unchecked_mut(i) = &*slice.get_unchecked(idx);
+            }
+            arr.assume_init()
+        }
```

Comment:
You can use `array::map` to simplify this:
```suggestion
        indices.map(|i| unsafe { &*slice.get_unchecked(i) })
```

Reply:
[`get_unchecked()`](https://doc.rust-lang.org/std/primitive.slice.html#method.get_unchecked) returns `&T`, so the `&*` would be unneeded. Apart from that - I started out with the `.map` function, but it generated worse code when I tried it. Believe me, I did not arrive at this complicated piece of unsafe code as a first choice. 😄